### ISSUE TITLE
Support for loki datasource running in multitenant mode

### ISSUE THREAD

**Issue by ialidzhikov** [grafana/grafana#16648]
**What would you like to be added**:
Grafana 6.1.3 Explore UI supports very well loki datasource running in single-tenant mode (with auth_enabled: false). But when loki is started with `auth_enabled: true` then Explore UI fails to list the labels - request to `api/datasources/proxy/1/api/prom/label?` returns `401 Unauthorized` with response body `no org id`.
The screenshot - ![alt text](https://i.imgur.com/GxqGU0h.png)
Basically I am interested how loki multitenancy is mapped to grafana multitenancy and whether there is already a roadmap about grafana support for multitenant loki.

**Why is this needed**:
Explore UI support for loki running in multitenant mode.

**Comment by ialidzhikov** [comment]
@davkal , maybe I miss some configuration?

**Comment by torkelo** [comment]
What auth credentials did you add when you configured the loki data source? Did you create loki tenant & credentials and entered those in the data source settings?

**Comment by ialidzhikov** [comment]
> What auth credentials did you add when you configured the loki data source?

Currently my datasource is the following:
```json
{
  "id": 1,
  "orgId": 1,
  "name": "Loki",
  "type": "loki",
  "typeLogoUrl": "public/app/plugins/datasource/loki/img/loki_icon.svg",
  "access": "proxy",
  "url": "http://loki:3100",
  "password": "",
  "user": "",
  "database": "",
  "basicAuth": true,
  "isDefault": true,
  "jsonData": {
     "keepCookies": [ ]
  },
  "readOnly": false
}
```
Also tried to set basic auth with username = value of `X-Scope-OrgID` header and the changes for username and password are not persisted to the datasource and they stay both "".

> Did you create loki tenant & credentials and entered those in the data source settings?

Hmm, I just run loki with `auth_enabled: true` and ingest the data with the `X-Scope-OrgID` header. For example a push request with curl for tenant `foo`:
```
$ curl -v -H "Content-Type: application/json" -H "X-Scope-OrgID: foo" -X POST http://loki:3100/api/prom/push -d '{"streams": [{"labels": "{app=\"foo\"}","entries": [{ "ts": "2019-04-17T18:51:32.889492872Z", "line": "foo" }]}]}'
```

Then I am able to fetch for my tenant `foo` with:
```
$ curl -v -H "Accept: application/json" -H "X-Scope-OrgID: ismo" -X GET http://loki:3100/api/prom/query?query=%7Bapp%3D%22foo%22%7D
```

**Comment by ialidzhikov** [comment]
Hmm, I cannot find occurrence of `X-Scope-OrgID` in the current master branch of grafana.
Shouldn't we put this in the roadmap for Explore UI (if it isn't already there)?

**Comment by torkelo** [comment]
You need some kind of auth proxy infront of loki to handle authentication (map basic auth to X-Scope-OrgID for example). Loki does not handle any authentication. 

You can set headers in provisioning. Not documented currently. 

```
{
  "jsonData": {
    "httpHeaderName1": "X-Scope-OrgID"
  },
  "secureJsonData":  {
    "httpHeaderValue1": "foo"
  }
}
```

**Comment by till** [comment]
@torkelo Thanks

---

So... in a multi-tenant environment. Where does this auth proxy come from?

I wonder how this works when I have one Grafana instance where I want to query all logs from my Loki server. They may have different "tenants", to shift through logs more easily, but I want a firehose of everything.

When I configure this header, I would then need to add one datasource per tenant (with the same loki) to be able to query it? I can't see how that's possible right now. Any thoughts/suggestions?

**Comment by torkelo** [comment]
Think you have to do all that yourself via some kind of proxy

**Comment by till** [comment]
@torkelo Can I set multiple org ids in one header?

**Comment by torkelo** [comment]
no

**Comment by sergiuspeian** [comment]
I have solved this issue, by configuring in [Loki Log Driver config](https://github.com/grafana/loki/blob/master/cmd/docker-driver/README.md#configure-the-default-logging-driver) file on */etc/docker/daemon.json*
 a **loki-tenant-id** value
```
{
    "debug" : true,
    "log-driver": "loki",
    "log-opts": {
        "loki-url": "https://loki.example.com/loki/api/v1/push",
        "loki-batch-size": "400",
        "loki-tenant-id": "myval"
    }
}
```
Then, on Grafana -> Data Sources - Loki:
**Custom HTTP Headers**:
*Header*: **X-Scope-OrgID**
*Value*: **myval**
![image](https://user-images.githubusercontent.com/29222087/79123809-a0415980-7da3-11ea-87e8-026129b8e978.png)

**Comment by aocenas** [comment]
So this seems to be solved already and working as expected so going to close this issue.

**Comment by CCOLLOT** [comment]
> > @torkelo Can I set multiple org ids in one header?
> 
> The documentation at https://grafana.com/docs/loki/latest/operations/multi-tenancy/ implies that multi-tenant-queries are possible by setting `X-Scope-OrgID: A|B` When I include a pipe `|` in the header (using [sergiuspeian](https://github.com/grafana/grafana/issues/16648#issuecomment-612899213)'s way) I get following error: `msg="Failed to call resource" error="multiple org IDs present`

Hi @the-voidl I face the same issue, did you find a solution?

**Comment by the-voidl** [comment]
no sorry @CCOLLOT. We decided to not use tenants in our deployment.
